src/corona_hakab_model/consts.py

```diff
@@ -219,8 +221,24 @@ class Consts(NamedTuple):
     ]
     should_isolate_positive_detected: bool = False
     isolate_after_num_day: int = 1  # will be in isolation the next day.
```

Comment:
done

Reply:
Code I tried, and did not seem to improve:
`    def progress_isolations(self):
        detected_positive_now = self.tested_positive_vector & \
                                (self.date_of_last_test == self.current_step)
        detected_positive_indices = np.flatnonzero(detected_positive_now)
        days_to_enter_isolation = self.consts.step_to_isolate_dist(size=len(detected_positive_indices))
        list_of_first_circle = [set.union(*[agent_connections.daily_connections for agent_connections in
                                            self.connection_data.connected_ids_by_strength[agent_index].values()])
                                for agent_index in detected_positive_indices]
        agents_in_first_circle = set.union(*list_of_first_circle) if len(list_of_first_circle) > 0 else set()
        daily_steps_to_isolate = self.consts.step_to_isolate_dist(size=len(agents_in_first_circle))
        for i, a in enumerate(agents_in_first_circle):
            if self.step_to_isolate_agent[a] < self.current_step:
                self.step_to_isolate_agent[a] = self.current_step + daily_steps_to_isolate[i]
        for agent_index, steps_to_isolate in \
                zip(detected_positive_indices, days_to_enter_isolation):
            # Get number of days agent is isolated
            if self.agents_in_isolation[agent_index] != IsolationTypes.NONE:
                # TODO: Need to take into account when got out of isolation also
                number_of_days_isolated = max(0,
                                              self.current_step - self.step_to_isolate_agent[agent_index])
            else:
                number_of_days_isolated = 0
            weekly_connection_isolation_ratio = 1 - min(number_of_days_isolated / 7, 1)
            for connected_agents in self.connection_data.connected_ids_by_strength[agent_index].values():
                #daily_steps_to_isolate = self.consts.step_to_isolate_dist(size=len(connected_agents.daily_connections))
                # All of this need to be home isolated
                #for daily_agent_id, daily_days_to_isolate in \
                #        zip(connected_agents.daily_connections, daily_steps_to_isolate):
                    # If were not isolated, and not tested positive now, needs to isolate
                #    if self.agents_in_isolation[daily_agent_id] == IsolationTypes.NONE and \
                #            not self.tested_positive_vector[daily_agent_id]:
                #        if self.step_to_isolate_agent[daily_agent_id] < self.current_step:
                #            self.step_to_isolate_agent[daily_agent_id] = self.current_step + \
                #                                                         daily_days_to_isolate
                agents_to_iterate = list(filter(lambda index: (self.agents_in_isolation[index] == IsolationTypes.NONE)
                                                              and not self.tested_positive_vector[index],
                                                connected_agents.weekly_connections))
                non_sick_agents = self.medical_machine.default_state_upon_infection.agents
                # We want to select which agents to isolate randomly. Starting with sick ones.
                shuffle(agents_to_iterate)
                # We want to first remove agents that are sick
                agents_to_iterate.sort(key=lambda index: self.agents[index] in non_sick_agents, reverse=True)
                how_many_to_isolate = int(round(len(agents_to_iterate) * weekly_connection_isolation_ratio))
                weekly_steps_to_isolate = self.consts.step_to_isolate_dist(size=how_many_to_isolate)
                for i in range(how_many_to_isolate):
                    # If already about to get isolated, do not update it
                    if self.step_to_isolate_agent[agents_to_iterate[i]] > self.current_step:
                        continue
                    self.step_to_isolate_agent[agents_to_iterate[i]] = self.current_step + weekly_steps_to_isolate[i]
            # Isolate the agent
            self.step_to_isolate_agent[agent_index] = self.current_step + steps_to_isolate

        # Isolating symptomatic agents
        for agent in self.new_agents_with_symptoms:
            # If is not getting ready to be isolated, or isolated already, then isolate
            if self.step_to_isolate_agent[agent_index] < self.current_step:
                self.step_to_isolate_agent[agent.index] = self.current_step + self.consts.step_to_isolate_dist
        # TODO: Remove healthy agents from isolation?
        self.isolate_agents()`

Reply:
I cannot remove the loop because I need to go over all the weekly.
Maybe the population was not big enough (20K), maybe on bigger one it can improve, on 20k it did not improve

Reply:
K